source/CIMenuScene.cpp

```diff
@@ -0,0 +1,391 @@
+//
+//  CIMenuScene.cpp
+//  CoreImpact
+//
+//  Created by Richard Yoon on 3/27/21.
+//  Copyright © 2021 Game Design Initiative at Cornell. All rights reserved.
+//
+
+#include "CIMenuScene.h"
+
+using namespace cugl;
+
+/** This is the ideal size of the logo */
+#define SCENE_SIZE  1024
+
+#pragma mark -
+#pragma mark Constructors
+
+/**
+ * Initializes the controller contents, making it ready for loading
+ *
+ * The constructor does not allocate any objects or memory.  This allows
+ * us to have a non-pointer reference to this controller, reducing our
+ * memory allocation.  Instead, allocation happens in this method.
+ *
+ * @param assets    The (loaded) assets for this game mode
+ *
+ * @return true if the controller is initialized properly, false otherwise.
+ */
+bool MenuScene::init(const std::shared_ptr<AssetManager>& assets) {
```

Comment:
A lot of these should be in a separate class. Maybe break it into something like settings scene, tutorial scene, lobby scene? The way this is et up right now will be extremely difficult to modify

Reply:
@kevinsun-dev do we need to have these all in the menu scene class? I agree that having each scene be its own class would simplify the code.